pkg/blackboxexporter/blackboxexporter.go

```diff
@@ -80,11 +89,17 @@ func (b *BlackBoxExporter) EnsureBlackBoxExporterDeploymentExists() error {
 		// populate the resource with the template
 		resource := populationFunc()
 		// and create it
-		err = b.Client.Create(b.Ctx, &resource)
-		if err != nil {
-			return err
-		}
+		return b.Client.Create(b.Ctx, &resource)
+	}
+
+	template := populationFunc()
+	if !b.Comparer.DeepEqual(resource.Spec, template.Spec) {
+		// template differs from resource, update to reconcile
+		b.Log.Info("the spec of the resource", "resource.Spec", resource.Spec)
+		resource.Spec = template.Spec
+		return b.Client.Update(b.Ctx, &resource)
```

Comment:
this fails because the resource has changed

Reply:
also, it will always be different after the first creation as there are default controllers that update deployments